tests/OtpTest.php

```diff
@@ -120,4 +120,45 @@ public function testHotpInvalidCounter()
 		$this->Otp->hotp($this->secret, 'a');
 	}
 
+	/**
+	 * Tests Otp->checkHotpResync()
+	 */
+	public function testHotpResync()
+	{
+		$secret = $this->secret;
+
+		// test default counter window
+		$this->assertEquals(0, $this->Otp->checkHotpResync($secret, 0, '755224'));
+		$this->assertEquals(1, $this->Otp->checkHotpResync($secret, 0, '287082'));
+		$this->assertEquals(2, $this->Otp->checkHotpResync($secret, 0, '359152'));
+
+		// test provided counter window
+		$this->assertEquals(3, $this->Otp->checkHotpResync($secret, 0, '969429', 3));
+		$this->assertEquals(4, $this->Otp->checkHotpResync($secret, 0, '338314', 4));
+		$this->assertEquals(5, $this->Otp->checkHotpResync($secret, 0, '254676', 5));
+
+		// test failures
+		$this->assertFalse($this->Otp->checkHotpResync($secret, 7, '287922'));
```

Comment:
Not 100% certain if assertFalse does type juggling, the docs don't say much, so maybe test it real quick. If it does and assertFalse(0) passes, then should also be "assertSame" here